tests/integration/golang/mlflow/namespace/flows/artifact_test.go

```diff
@@ -0,0 +1,284 @@
+//go:build integration
+
+package flows
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api/request"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api/response"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/common"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/helpers"
+)
+
+type ArtifactFlowTestSuite struct {
+	suite.Suite
+	helpers.BaseTestSuite
+	testBuckets []string
+	s3Client    *s3.Client
+}
+
+// TestArtifactFlowTestSuite tests the full `artifact` flow connected to namespace functionality.
+// Flow contains next endpoints:
+// - `GET /artifacts/get`
+// - `GET /artifacts/list`
+func TestArtifactFlowTestSuite(t *testing.T) {
+	suite.Run(t, &ArtifactFlowTestSuite{
+		testBuckets: []string{"bucket1", "bucket2"},
+	})
+}
+
+func (s *ArtifactFlowTestSuite) SetupTest() {
+	s.BaseTestSuite.SetupTest(s.T())
+
+	s3Client, err := helpers.NewS3Client(helpers.GetS3EndpointUri())
+	assert.Nil(s.T(), err)
+	s.s3Client = s3Client
+}
+
+func (s *ArtifactFlowTestSuite) TearDownTest() {
+	assert.Nil(s.T(), s.NamespaceFixtures.UnloadFixtures())
+}
+
+func (s *ArtifactFlowTestSuite) Test_Ok() {
+	tests := []struct {
+		name           string
+		setup          func() (*models.Namespace, *models.Namespace)
+		namespace1Code string
+		namespace2Code string
+	}{
+		{
+			name: "TestInScopeOfTwoCustomNamespaces",
+			setup: func() (*models.Namespace, *models.Namespace) {
+				return &models.Namespace{
+						Code:                "namespace-1",
+						DefaultExperimentID: common.GetPointer(int32(0)),
+					}, &models.Namespace{
+						Code:                "namespace-2",
+						DefaultExperimentID: common.GetPointer(int32(0)),
+					}
+			},
+			namespace1Code: "namespace-1",
+			namespace2Code: "namespace-2",
+		},
+		{
+			name: "TestInScopeOfOneDefaultAndOneCustomNamespacesObviousCase",
```

Comment:
i think theses names are hard to read because so long -- maybe "TestDefaultAndCustom", "TestImplicitAndCustom", etc?

Reply:
fixed.